pkg/providers/securitygroup/securitygroup.go

```diff
@@ -55,7 +55,13 @@ func (p *Provider) List(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTempl
 	p.Lock()
 	defer p.Unlock()
 	// Get SecurityGroups
-	securityGroups, err := p.getSecurityGroups(ctx, p.getFilters(nodeTemplate))
+	// TODO: When removing custom launchTemplates for v1beta1, security groups will be required.
+	// The check will not be necessary
+	filters := p.getFilters(nodeTemplate)
+	if len(filters) == 0 {
```

Comment:
This seems reasonable but I would expect the same behavior for subnet provider as well; otherwise, the divergent behavior is going to get confusing